### PR TITLE
Add support for arm64 builds

### PR DESCRIPTION
**Pull request by jalaziz**
Rather than forcing nitriding to build against the amd64 architecture, support arm64 builds by allowing the `ARCH` environment variable to override the build architecture.

Also, update the example to automatically choose the build architecture based on the architecture of the machine where the example is built and run.

### CODE REVIEW

File: example/Makefile [hunk 1]
```diff
@@ -13,12 +21,15 @@ image: $(image_tar)
 $(image_tar): Dockerfile service.py start.sh
 	docker run \
 		-v $(PWD):/workspace \
-		gcr.io/kaniko-project/executor:v1.9.2 \
+		gcr.io/kaniko-project/executor:v1.19.2 \
```
Comment (rillian) [inline example/Makefile:24]: is this bump necessary for the arch changes? If not, please move it to a separate PR.

Comment (jalaziz) [reply example/Makefile]: I thought it was needed, but just tested again and seems to work with the old version. I've reverted.

**Review by rillian — APPROVED** [review]
Looks reasonable, to me. Thanks!
